### ISSUE TITLE
Expose `ResourceFormatLoader`'s virtual "exists()" function to GDScript

### ISSUE THREAD

**Issue by cgbeutler** [godotengine/godot-proposals#2772]
### Describe the project you are working on
GDScript Plugin that loads from virtual paths.

### Describe the problem or limitation you are having in your project
I would like to use virtual paths to load resources that don't directly exist on the filesystem. This is doable in C++. Sadly, the virtual `exists()` function is not exposed to GDScript. Instead, you are left to the default `exists` which simply checks for a file on the filesystem. (Link: [ResourceFormatLoader::exists](https://github.com/godotengine/godot/blob/af03e9c83018b47cb82da67e5f8e277b29b82d66/core/io/resource_loader.cpp#L99-L101))

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
ResourceFormatLoader exposes most of the virtual functions within to GDScript. Exposing `exists` as well would be all that was needed to allow virtual paths.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
Pretty sure it's a straightforward one, since the C++ is already so close. Basically just exposing the method.
As a result, ResourceFormatLoader would allow you to declare `exists`. Replicating the default behavior would be something like:
```gdscript
func exists( path: String ):
  var file := File.new()
  return file.exists( path )
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?
Not overriding the exists function would give you the default file check.

### Is there a reason why this should be core and not an add-on in the asset library?
I don't think exposing an additional method can happen in an add-on

**Comment by fire** [comment]
Can you list a full example of a virtual resource?

**Comment by Calinou** [comment]
@fire Virtual paths are generally used for network abstraction, or for virtual filesystems (e.g. PCK files).

**Comment by cgbeutler** [comment]
Yeah, Godot has almost no real paths. Everything in res and import becomes a virtual path on export to a pack file!

As for more specifics on what I'm building, I want to write a res-format-loader where you can load "profile://path" and get back _either_ the base file found in res://_db/path or the updated save version from user://_db/path automagically. Everything works fine except "exists"
